ui/perfherder/compare/CompareTable.jsx

```diff
@@ -36,33 +138,66 @@ export default class CompareTable extends React.PureComponent {
       >
         <thead>
           <tr className="subtest-header bg-lightgray">
-            <th className="text-left">
-              <span>{testName}</span>
-              {onPermalinkClick && (
-                <Button
-                  className="permalink p-0 ml-1"
-                  color="link"
-                  onClick={() =>
-                    onPermalinkClick(
-                      getHashBasedId(testName, hashFunction),
-                      history,
-                      this.header,
-                    )
-                  }
-                  title="Permalink to this test table"
-                  aria-label={`Permalink to test ${testName} table`}
-                >
-                  <FontAwesomeIcon icon={faHashtag} />
-                </Button>
-              )}
-            </th>
-            <th className="table-width-lg">Base</th>
-            {/* empty for less than/greater than data */}
-            <th className="table-width-sm" aria-label="Comparison" />
-            <th className="table-width-lg">New</th>
-            <th className="table-width-lg">Delta</th>
-            <th className="table-width-lg">Magnitude of Difference</th>
-            <th className="table-width-lg">Confidence</th>
+            {columns.map(({ name, currentSort }, index) => {
```

Comment:
This mapping is a bit difficult to comprehend. We seem to need it because it gives as an index as a form of identifying the column to sort by. I think that by changing the `this.state.columns` data structure from an array to a key-value map we can restore the old way of defining the `<th/>`s.